recipe-app/src/pages/app/mealPlans/add/addMealPlan/AddMealPlan.tsx

```diff
@@ -1,5 +1,93 @@
+import { useFormik } from "formik";
+import { useEffect, useState } from "react";
+import { recipeApi } from "../../../../../api/recipes";
+import { Recipe } from "../../../../../types/editRecipe";
+import { MealPlan } from "./types";
+import MealTable from "./components/mealTable/MealTable";
+import { DayName } from "../../../../../types/MealPlan";
+
+import "./addMealPlan.scss";
+import DataInput from "./components/dataInput/DataInput";
+import PlanInput from "./components/planInput/PlanInput";
+import PlanTextArea from "./components/planTextArea/PlanTextArea";
+
 function AddMealPlan() {
-	return <p>Add Meal Plan</p>;
-}
+	const [recipes, setRecipes] = useState<Recipe[]>([]);
+	useEffect(() => {
+		async function fetchRecipes() {
+			const recipeList = await recipeApi.getAll();
+			setRecipes(recipeList);
+		}
+
+		fetchRecipes();
+	}, []);
+
+	const onSubmit = async (values: MealPlan) => {
+		try {
+			console.log({ values });
+
+			// @ts-expect-error do poprawy typowanie recipeApi.addMealPlan
+			await recipeApi.addMealPlan(values);
+			alert("Plan posiłków został zapisany!");
+		} catch (error) {
+			alert("Wystąpił błąd przy zapisywaniu planu.");
+		}
+	};
 
+	const formik = useFormik<MealPlan>({
+		initialValues: {
+			name: "",
+			description: "",
+			dateFrom: "",
+			mealName: [],
```

Comment:
```suggestion
			plan: {},
```

Dodatkowo poprawnie otypuj